goth/api_monitor/router_addon.py

```diff
@@ -50,11 +55,10 @@ def request(self, flow: HTTPFlow) -> None:
             node_name = self._node_names[remote_addr]
 
             if server_port == YAGNA_REST_PORT:
-                # It's a provider agent calling a yagna daemon.
-                # We assume that both are running on the same host, so the
-                # request is bounced back to the caller.
-                req.host = remote_addr
-                req.port = YAGNA_REST_PORT
+                # It's a provider agent calling a yagna daemon
+                # We route the request to the daemon's mapped port on the host
```

Comment:
I think it's still worth mentioning what we assume about the agent and daemon running on the same host (it explains why we're routing the request like this).